lib/active_record_encryption/encryptor/registry.rb

```diff
@@ -2,32 +2,15 @@
 
 module ActiveRecordEncryption
   module Encryptor
-    class Registry
-      def initialize
-        @registrations = []
-      end
-
-      def register(encryptor_name, klass = nil, **options, &block)
-        block ||= proc { |_, *args| klass.new(*args) }
-        registrations << Registration.new(encryptor_name, block, **options)
-      end
-
-      def lookup(symbol, *args)
-        registration = find_registration(symbol, *args)
-
-        if registration
-          registration.call(self, symbol, *args)
-        else
-          raise ArgumentError, "Unknown encryptor #{symbol.inspect}"
-        end
-      end
-
+    class Registry < ActiveModel::Type::Registry
       private
 
-      attr_reader :registrations
+      def registration_klass
+        Registration
+      end
 
-      def find_registration(symbol, *args)
-        registrations.find { |registration| registration.matches?(symbol, *args) }
+      def find_registration(symbol, *args, **kwargs)
+        registrations.find { |r| r.matches?(symbol, *args, **kwargs) }
```

Comment:
Actually this should be fixed in `ActiveModel::Type::Registry` but it is missing in Rails 6.0.3.
It will be fixed in Rails 6.0.4 https://github.com/rails/rails/commit/a496537e364d76798683c921664d2f48b90fef5c.